### PR TITLE
replace path with dataset for select_agents, avoid calling open again in AgentDataset

### PR DESCRIPTION
**Pull request by lucabergamini**
Replace first argument of select_agents (Path) with the `ChunkedStateDataset` itself. This avoid performing an `open()` call when `AgentDataset` builds the mask at runtime.

Note this is slightly related to #68. When both will be merged the behaviour will be back to normal (plus select_agents will be more accurate)

### CODE REVIEW

File: l5kit/l5kit/dataset/agent.py [hunk 1]
```diff
@@ -52,17 +52,17 @@ def load_agents_mask(self) -> np.ndarray:
...
                 history_num_frames,
                 future_num_frames,
                 th_yaw_degree=TH_YAW_DEGREE,
                 th_extent_ratio=TH_EXTENT_RATIO,
                 th_movement=TH_MOVEMENT,
                 th_distance_av=TH_DISTANCE_AV,
-                num_workers=8,
-            )  # TODO maybe set in env var?
-            self.dataset.open()  # ensure root is updated
+                num_workers=8,  # TODO maybe set in env var?
```
Comment (yye-lyft) [inline l5kit/l5kit/dataset/agent.py:63]: How about remove it here. Just use the `cpu_count` that `multiprocessing package` gives you for the paralleisation in `select_agents.py`?
`from multiprocessing import cpu_count, Pool`
...
`with Pool(cpu_count()) as pool:`

File: l5kit/l5kit/dataset/select_agents.py [hunk 2]
```diff
@@ -266,8 +262,11 @@ def select_agents(
     args = parser.parse_args()
 
     for input_folder in args.input_folder:
```
Comment (yye-lyft) [inline l5kit/l5kit/dataset/select_agents.py:264]: `args.input_folders`?

File: l5kit/l5kit/dataset/agent.py [hunk 3]
```diff
@@ -52,17 +52,17 @@ def load_agents_mask(self) -> np.ndarray:
...
+            )
+            self.dataset.open()  # ensure root is updated, reload all arrays TODO lberg: avoid reload
```
Comment (yye-lyft) [inline l5kit/l5kit/dataset/agent.py:65]: why we need to call `open` again?

Comment (lucabergamini) [reply l5kit/l5kit/dataset/agent.py:65]: to update the root dict basically (otherwise you get a KeyError again). Let me see if I can find a better way

Comment (yye-lyft) [reply l5kit/l5kit/dataset/agent.py:64]: it can be in a subsequent PR, I will approve it for now

**Pull request merged**